db/migrate/20220326101723_create_join_table_user_post.rb

```diff
@@ -0,0 +1,8 @@
+class CreateJoinTableUserPost < ActiveRecord::Migration[6.1]
+  def change
+    create_join_table :users, :posts, column_options: { null: false }, table_name: :likes do |t|
+      t.index :user_id, unique: true
+      t.index :post_id, unique: true
+    end
+  end
+end
```

Comment:
- [ ] Kindly create the `likes` table just as you created other tables because according to the requirement, `likes` is a table on its own having a timestamp and referencing post, and `author` in the `users` table. See how the `likes` table should look like in the migration file;

```ruby
  def change
    create_table :likes do |t|
      t.timestamps
    end
  end
```

- [ ] Kindly generate migrations for adding post reference and user reference to likes table instead of creating a join table as this is part of the requirement. Generate the migrations just as you did with the `comments` and remember to use the hint I gave you. After successfully creating the `likes` table, the likes table in your schema file should look close to this 👇 

```ruby
 create_table "likes", force: :cascade do |t|
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
    t.bigint "post_id"
    t.bigint "author_id"
    t.index ["author_id"], name: "index_likes_on_author_id"
    t.index ["post_id"], name: "index_likes_on_post_id"
  end
```